src/Microsoft.Extensions.OData.Migration/MigrationOptions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Microsoft.Extensions.OData.Migration
+{
+    /// <summary>
+    /// Container for parameters to pass into middleware.
+    /// </summary>
+    public sealed class MigrationOptions
+    {
+        // Service root (e.g. the https://foo:80/odata part of https://foo:80/odata/Products) is required for parser
+        public Uri ServiceRoot { get; set; }
+
+        // V4 model to validate request
+        public Microsoft.OData.Edm.IEdmModel V4Model { get; set; }
+
+        // Path to V3 model edmx; pass in as EDMX because if V3 model is required then 

```

Comment:
What if we passed the model itself instead of the path to the EDMX to the UseMigrationMiddleware? They can derive the model themselves, or create it from the EDMX.

Reply:
How about we support both?

Reply:
Changed to be models being passed in for consistency - will add edmx option later